lib/engine/alerts/api_fetcher.ex

```diff
@@ -101,18 +101,46 @@ defmodule Engine.Alerts.ApiFetcher do
     end
   end
 
-  @spec stops_for_alert(alert()) :: [Fetcher.stop_id()]
-  defp stops_for_alert(alert) do
+  @spec stops_for_alert(alert(), StationConfig.t()) :: [Fetcher.stop_id()]
+  defp stops_for_alert(alert, station_config) do
     alert["attributes"]["informed_entity"]
     |> Enum.flat_map(fn ie ->
       if ie["stop"] do
         [ie["stop"]]
       else
-        []
+        # Route-level alerts aren't associated w/ a specific stop_id,
+        # but should apply to every stop_id on the route
+        case ie["route"] do
+          nil ->
+            []
+
+          route ->
+            stop_ids_for_route(route, station_config)
+        end
       end
     end)
   end
 
+  @spec stop_ids_for_route(String.t(), StationConfig.t()) :: [String.t()]
+  defp stop_ids_for_route(route_id, station_config) do
+    route_ids =
+      case route_id do
+        "Green-E" ->
+          ["Green-E", "Green-E-glx"]
+
+        "Green" ->
```

Comment:
This will never be the value of the `route` field of an informed entity, because it is not a route ID that exists. If multiple Green Line branches were affected there would have to be one entity with `route: "Green-B"`, a second entity with `route: "Green-C"`, etc.

Reply:
I'm not sure what you mean by restructuring? This seems like it should fit right into the existing structure you have.

```elixir
"Green-B" -> ~w[...]
"Green-C" -> ~w[...]
"Green-D" -> ~w[...]
"Green-E" -> ~w[Green-E Green-E-glx]
```

Reply:
Oh I misunderstood your explanation! Yes that would fit in perfectly if they're all in the same informed entity. I read that as each branch would have its own entity. Thanks for pointing this out

Reply:
In my haste to respond, I did not type my thoughts correctly. Edited my previous comment 😅

Reply:
Ah, okay, I am un-confused. And yes I'm pretty sure this would all be within the same alert.

Reply:
I would still like to add a unit test case for the Green Line, but I've updated the logic to include alerts that affect the entire Green Line. Also restructured things so it's a bit easier to understand I think.